main/src/ca/uwaterloo/flix/util/ParOps.scala

```diff
@@ -17,30 +17,38 @@
 package ca.uwaterloo.flix.util
 
 import java.util
-import java.util.concurrent.{Callable, Executors}
-
+import java.util.concurrent.{Callable, Future}
 import scala.jdk.CollectionConverters._
-import scala.collection.parallel._
 import ca.uwaterloo.flix.api.Flix
 
+import scala.collection.mutable.ArrayBuffer
+
 object ParOps {
 
   /**
     * Apply the given function `f` to each element in the list `xs` in parallel.
     */
   @inline
   def parMap[A, B](xs: Iterable[A])(f: A => B)(implicit flix: Flix): Iterable[B] = {
-    // Build the parallel array.
-    val parArray = xs.toParArray
-
-    // Configure the task support.
-    parArray.tasksupport = flix.forkJoinTaskSupport
+    val out = ArrayBuffer.fill(xs.size)(null.asInstanceOf[B])
+
+    val futuresBuilder = ArrayBuffer.newBuilder[Future[?]]
+    futuresBuilder.sizeHint(xs.size)
+    val futures = futuresBuilder.result()
```

Comment:
OK; so I understand why this is done. I just wonder if we could do better.

In particular, if we can avoid allocation of `n` `Future`s.

It seems one option would be to use https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#awaitTermination-long-java.util.concurrent.TimeUnit-

But the problem is that it shutsdown the threadpool. Then we would need to create a new thread pool, but that is very expensive.

So I am not sure if it can be done better than what you have?

Reply:
Maybe it would be more efficient to use a CountDownLatch:

https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/util/concurrent/CountDownLatch.html

Reply:
Does it make sense? I am trying to reduce allocation of `Future`s.